File: PythonDemo/src/demo/SimpleExampleModified_1.py
# -*- coding:UTF-8 -*-

'''
Created on 2012-12-18

@author: chengkai
'''

from selenium import webdriver
from selenium.webdriver.support.ui import WebDriverWait 
from selenium.common.exceptions import ElementNotVisibleException
import time



def main():
    try:
#        driver= webdriver.Chrome()
        driver= webdriver.Firefox()
        
        driver.get('http://www.360buy.com/')
     
        WebDriverWait(driver, 20, 1, (ElementNotVisibleException)).until
        (lambda driber: driver.find_element_by_id('subscription-submit').is_displayed())
#     by id
        input_search = driver.find_element_by_id('key');
#     by xpath
        button_search = driver.find_element_by_xpath("//*[@id='search']//input[@type='button']")
#     by class name  
        branch_address_1 = driver.find_element_by_class_name("ar")
        print 'by class name: '+branch_address_1.text
#     by css selector
        branch_address_2 = driver.find_element_by_css_selector("p.ar")
        print 'by css selector: '+branch_address_2.text
#     by link text  
        about_us = driver.find_element_by_link_text('关于我们')
        print 'by link text: ' + about_us.text
#     by tag name  
        jd_report = driver.find_element_by_id("jdnews").find_element_by_tag_name("h2")
        print 'by tag name: ' + jd_report.text
        
        input_search.send_keys("12.12")
        button_search.click()
      
        WebDriverWait(driver, 20, 1, (ElementNotVisibleException)).until
        (lambda driver: driver.find_element_by_xpath("//*[@id='plist']").is_displayed())
    except:
        print 'has an error!'
        raise
    finally:
        driver.quit()

if __name__ == '__main__':
    main()
    

File: SeleniumDemo/src/noframe/cases/SimpleExampleModified_1.java
package noframe.cases;

import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedCondition;
import org.openqa.selenium.support.ui.WebDriverWait;

public class SimpleExampleModified_1 {
	public static void main(String[] args) {
		WebDriver driver=  new FirefoxDriver();
        driver.get("http://www.360buy.com/");
        
        (new WebDriverWait(driver, 20)).until(new ExpectedCondition<Boolean>() {
            public Boolean apply(WebDriver d) {
                return d.findElement(By.id("subscription-submit")).isDisplayed();
            }
        });
       
        //by id
        WebElement input_search = driver.findElement(By.id("key")); 
        
        // by xpath
        WebElement button_search = driver.findElement(By.xpath("//*[@id='search']//input[@type='button']"));
        
        //by class name
        WebElement branch_address_1 = driver.findElement(By.className("ar"));
        System.out.println("by class name: "+branch_address_1.getText());
        
        // by css selector
        WebElement branch_address_2 = driver.findElement(By.cssSelector("p.ar"));
        System.out.println("by css selector: "+branch_address_2.getText());
        
        // by link text
        WebElement about_us = driver.findElement(By.linkText("关于我们"));
        System.out.println("by link text: "+about_us.getText());
        
        //by tag name
        WebElement jd_report = driver.findElement(By.id("jdnews")).findElement(By.tagName("h2"));
        System.out.println("by tag name: "+jd_report.getText());
        
        input_search.sendKeys("12.12");

        button_search.click();

        (new WebDriverWait(driver, 20)).until(new ExpectedCondition<Boolean>() {
            public Boolean apply(WebDriver d) {
                return d.findElement(By.id("plist")).isDisplayed();
            }
        });
        
        List<WebElement> products_name = driver.findElements(By.xpath("//div[@id='plist']//li/div[2]"));
        for(WebElement product: products_name){
//        	System.out.println(product.getText());
        }
        driver.quit();
	}
	
}
